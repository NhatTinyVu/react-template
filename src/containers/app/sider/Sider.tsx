import React from 'react';
import { Link } from 'react-router-dom';
import { Layout, Menu } from 'antd';
import { routes } from '../constants';

const Sider: React.FC = () => (
  <Layout.Sider>
    <Menu mode="inline" defaultSelectedKeys={[routes[0].url]}>
      {routes.map(({ title, url, childrens }) => (
        <Menu.SubMenu key={url} title={<Link to={url}>{title}</Link>}>
          {childrens.map(route => (
            <Menu.Item key={route.url}>{route.title}</Menu.Item>
          ))}
        </Menu.SubMenu>
      ))}
    </Menu>
  </Layout.Sider>
);

export default Sider;
