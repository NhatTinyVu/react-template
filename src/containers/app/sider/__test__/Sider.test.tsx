import React from 'react';
import { shallow } from 'enzyme';
import * as useRouter from 'use-react-router';
import toJSON from 'enzyme-to-json';
import { Item } from '../Sider';
import Sider from '..';

jest
  .spyOn(useRouter, 'default')
  .mockImplementation((): any => ({ location: { pathname: '/react-router' } }));

describe('Sider', () => {
  test('renders without crashing', (): void => {
    const wrapper = shallow(<Sider />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders correctly', (): void => {
    const wrapper = shallow(<Item url="url" title="title" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
