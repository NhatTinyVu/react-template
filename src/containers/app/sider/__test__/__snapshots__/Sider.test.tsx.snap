// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sider renders correctly 1`] = `
<span>
  <Icon
    type="unordered-list"
  />
  <span>
    <Link
      to="url"
    >
      title
    </Link>
  </span>
</span>
`;

exports[`Sider renders without crashing 1`] = `
<Sider
  breakpoint="md"
  collapsedWidth="0"
>
  <Menu
    defaultSelectedKeys={
      Array [
        "/react-hooks",
      ]
    }
    mode="inline"
    selectedKeys={
      Array [
        "/react-hooks",
      ]
    }
    theme="dark"
  >
    <SubMenu
      key="/react-hooks"
      title={
        <Item
          title="React hooks"
          url="/react-hooks"
        />
      }
    >
      <MenuItem
        key="/react-hooks/use-event-listener"
      >
        <Link
          to="/react-hooks/use-event-listener"
        >
          useEventListener
        </Link>
      </MenuItem>
      <MenuItem
        key="/react-hooks/use-why-did-you-update"
      >
        <Link
          to="/react-hooks/use-why-did-you-update"
        >
          useWhyDidYouUpdate
        </Link>
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;
